src/corenetwork/lteip/IP2lte.cc

```diff
@@ -175,27 +212,35 @@ void IP2lte::toStackUe(IPv4Datagram * datagram)
         seqNums_.insert(p);
     }
 
-    int headerSize = datagram->getHeaderLength();
+    int headerSize = B(ipHdr->getHeaderLength()).get();
 
     // inspect packet depending on the transport protocol type
-    switch (transportProtocol)
-    {
-        case IP_PROT_TCP:
-            inet::tcp::TCPSegment* tcpseg;
-            tcpseg = check_and_cast<inet::tcp::TCPSegment*>(transportPacket);
-            srcPort = tcpseg->getSrcPort();
-            dstPort = tcpseg->getDestPort();
-            headerSize += tcpseg->getHeaderLength();
+    // TODO: needs refactoring (redundant code, see toStackEnb())
+    switch (transportProtocol) {
+        case IP_PROT_TCP: {
+            auto& tcpHdr = datagram->peekAtFront<tcp::TcpHeader>(b(-1),
+                    Chunk::PF_ALLOW_SERIALIZATION);
```

Comment:
I think this PF_ALLOW_SERIALIZATION is not needed here. I guess it's just a leftover from the migration process. This flag allows serialization between arbitrary headers. Binary to/from TcpHeader is allowed by default.

Reply:
Yes, correct. Removed the B(-1) and the flag.